httpx/config.py

```diff
@@ -241,6 +250,10 @@ def __init__(
                 self.connect_timeout = timeout.connect_timeout
                 self.read_timeout = timeout.read_timeout
                 self.write_timeout = timeout.write_timeout
+            elif isinstance(timeout, UnsetType):
```

Comment:
Is there any reason (maybe type hints-related?) we can't use `elif timeout is UNSET` here?

Reply:
I tried that first but `mypy` was complaining, I will reproduce the actual error once I get home, maybe there is something I missed.

Reply:
This is the mypy error:

```
httpx/config.py:262: error: Incompatible types in assignment (expression has type "Union[float, UnsetType]", variable has type "Optional[float]")
httpx/config.py:263: error: Incompatible types in assignment (expression has type "Union[float, UnsetType]", variable has type "Optional[float]")
httpx/config.py:264: error: Incompatible types in assignment (expression has type "Union[float, UnsetType]", variable has type "Optional[float]")
```

The weird thing is that it complains about these lines:
```
else:
    self.connect_timeout = timeout
    self.read_timeout = timeout
    self.write_timeout = timeout
```

Reply:
emm, this is because you declare the type of `timeout` is `TimeoutTypes`(in `__init__` method). It's a `Union` which includes `UnsetType`. And  you declare the type of  `self.connect_timeout` is `typing.Optional[float]` in line 241. So if you use `elif timeout is UNSET`, any instance of `UnsetType` will enter the `else` branch.